### ISSUE TITLE
AN Blender 2.80 Beta: bug when instancing object inside and outside the Collection

### ISSUE THREAD

**Issue by enzyme69** [JacquesLucke/animation_nodes#962]
When I am changing instance inside and outside collection, sometimes it is giving error message and the node tree failed to run. This is new.

**Comment by OmarSquircleArt** [comment]
What is the error? What exactly are you doing?
Are you using the object instancer node with the source as an object that belongs to a collection?

**Comment by enzyme69** [comment]
Let say a scene as Icosphere and all instance objects all working fine, opened from previous Blender file. Then I created a Torus and pick the instance. And it does not work.

**Comment by OmarSquircleArt** [comment]
You have a blend file with an object instancer node that instances an icosphere. If you open that file and change the source object to a torus it doesn't work? Are there any errors?

**Comment by enzyme69** [comment]
It works if everything is outside collection, but it crashes sometimes.

```

Error: animation channel (index = 1) not found in mouse_action_keys()
Info: Deleted 2 object(s)

Info: Deleted 1 object(s)

Info: Total files 0 | Changed 0 | Failed 0
Info: Saved "ico_torus_001.blend"

Info: Saved "ico_torus_001.blend"

Info: Saved "ico_torus_001.blend"

Info: Saved "ico_torus_002.blend"

Read blend: /Users/jimmygunawan/Desktop/ico_torus_002.blend
Writing: /tmp/ico_torus_002.crash.txt
Segmentation fault: 11
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]


```

**Comment by enzyme69** [comment]
In previous Blender, all those instances are nicely parented under Empty. What's the plan for 2.80?

**Comment by enzyme69** [comment]
Also "toggle viewport" button under Instance node does not work.
```

location: <unknown location>:-1
Traceback (most recent call last):
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/operators/dynamic_operators.py", line 52, in invoke_InvokeFunction
    return self.execute(context)
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/operators/dynamic_operators.py", line 58, in execute_InvokeFunction
    self.an_executeCallback(self.callback, *args)
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/extend_bpy_types.py", line 28, in _executeCallback
    executeCallback(callback, *args, **kwargs)
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/operators/callbacks.py", line 25, in executeCallback
    callback(*parameters, args, kwargs)
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/operators/callbacks.py", line 48, in executeSocketCallback
    getattr(socket, functionName)(*args, **kwargs)
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/sockets/object.py", line 67, in toggleObjectVisibilty
    object.hide_viewport = not object.hide_viewport
AttributeError: type object 'object' has no attribute 'hide_viewport'
```

**Comment by enzyme69** [comment]
But the biggest issue remains like this:
By default, instances are outside of Collections. If I made them goes inside the Collection, together with their master instance object, then I replace the master with a new object, I am getting error and node tree broken.


```

Traceback (most recent call last):
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/execution/main_execution_unit.py", line 38, in executeUnit
    exec(self.executeCodeObject, self.executionData, self.executionData)
  File "execution: 'NodeTree'", line 11, in <module>
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/nodes/object/object_instancer.py", line 129, in getInstances_WithSource
    self.removeAllObjects()
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/nodes/object/object_instancer.py", line 208, in removeAllObjects
    self.removeObject(object)
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/nodes/object/object_instancer.py", line 228, in removeObject
    self.unlinkInstance(object)
  File "/Users/jimmygunawan/Library/Application Support/Blender/2.80/scripts/addons/animation_nodes/nodes/object/object_instancer.py", line 326, in unlinkInstance
    scene.collection.objects.unlink(object)
RuntimeError: Error: Object 'Torus.001' not in collection 'Master Collection'


```

**Comment by OmarSquircleArt** [comment]
We will not be looking into any crashes because they might be caused by blender.
The toggle visibility operator was fixed in  66cbdd8.

Will look into the instancer node.

**Comment by OmarSquircleArt** [comment]
Fixed in  35c1a18.
